internal/provider/trust_provider_resource.go

```diff
@@ -110,7 +110,30 @@ func (r *trustProviderResource) Schema(_ context.Context, _ resource.SchemaReque
 				},
 			},
 			"aws_ecs_role": schema.SingleNestedAttribute{
-				Description: "AWS ECS Role type Trust Provider configuration.",
+				Description:        "AWS ECS Role type Trust Provider configuration.",
+				Optional:           true,
+				DeprecationMessage: "Deprecated. Use aws_role instead",
+				Attributes: map[string]schema.Attribute{
+					"account_id": schema.StringAttribute{
+						Description: "The ID of the AWS account that is hosting the ECS Task.",
+						Optional:    true,
+					},
+					"assumed_role": schema.StringAttribute{
+						Description: "The Name of the AWS IAM Role which is running the ECS Task.",
+						Optional:    true,
+					},
+					"role_arn": schema.StringAttribute{
+						Description: "The ARN of the AWS IAM Role which is running the ECS Task.",
+						Optional:    true,
+					},
+					"username": schema.StringAttribute{
+						Description: "The UsernID of the AWS IAM Account which is running the ECS Task (not commonly used).",
+						Optional:    true,
+					},
+				},
+			},
+			"aws_role": schema.SingleNestedAttribute{
+				Description: "AWS Role type Trust Provider configuration.",
 				Optional:    true,
 				Attributes: map[string]schema.Attribute{
 					"account_id": schema.StringAttribute{
```

Comment:
You have several references to ECS in the descriptions here which need to be cleaned up.

Reply:
Where can I get the correct descriptions for aws_role type attributes?